### ISSUE TITLE
Value based Row mapping throws NullPointerException even if row value cannot be null

### ISSUE THREAD

**Issue by vinkovorih** [jOOQ/jOOQ#13606]
### Expected behavior
RecordMapper should map name and surname entities because in select query we are omitting all non null entries persisted in database.

### Actual behavior
```
Caused by: java.lang.NullPointerException: name is marked @NonNull but is null
        at org.test.TestPojo.<init>(TestPojo.java:4) ~[test.jar:0.0.1]
        at org.jooq.Records.lambda$mapping$22(Records.java:695) ~[jooq-3.15.4.jar:?]
        at org.jooq.RecordMapper.apply(RecordMapper.java:80) ~[jooq-3.15.4.jar:?]
        at org.jooq.RecordMapper.apply(RecordMapper.java:65) ~[jooq-3.15.4.jar:?]
        at org.jooq.Converter$1.from(Converter.java:178) ~[jooq-3.15.4.jar:?]
        at org.jooq.impl.DefaultBindingGetResultSetContext$1.value(DefaultBindingGetResultSetContext.java:105) ~[jooq-3.15.4.jar:?]
        at org.jooq.impl.DefaultBinding$AbstractBinding.get(DefaultBinding.java:951) ~[jooq-3.15.4.jar:?]
        at org.jooq.impl.ChainedConverterBinding.get(ChainedConverterBinding.java:97) ~[jooq-3.15.4.jar:?]
        at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1551) ~[jooq-3.15.4.jar:?]
        at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1500) ~[jooq-3.15.4.jar:?]
```

### Steps to reproduce the problem

```
dslContext.select(row(TEST.NAME,
                      TEST.SURNAME).mapping(TestPojo::new))
                        .from(TEST)
			.where(TEST.NAME.isNotNull());
```

```
@Value
@Builder
public class TestPojo {
	@NonNull
	private final String name;
	private final String surname
}
```

### Versions

- jOOQ: 3.15.4
- Java: 11.0.13 Temurin
- Database (include vendor): Postgres 13.5
- OS: Centos7
- JDBC Driver (include name if inofficial driver): postgresql-42.3.1

**Comment by lukaseder** [comment]
A lot of things have been fixed in this area. Can you please try the latest 3.15.x version, which is 3.15.10:
https://www.jooq.org/download/versions

Or even the latest 3.16.6 version, in case a minor release update is viable

**Comment by vinkovorih** [comment]
Thank you for a fast response. It seems that issue was resolved with latest 3.15.x version.
